src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php

```diff
@@ -108,13 +108,13 @@ public static function getServiceProvider(ContainerBuilder $container, $name)
     {
         $container->resolveEnvPlaceholders($name, null, $usedEnvs);
 
-        if (0 === strpos($name, 'redis://') || $usedEnvs) {
+        if ($usedEnvs || preg_match('#^[a-z]++://#', $name)) {
             $dsn = $name;
 
             if (!$container->hasDefinition($name = md5($dsn))) {
-                $definition = new Definition(\Redis::class);
+                $definition = new Definition('_');
```

Comment:
What about using `AdapterInterface::class` to not make it harder for bundles that do static analyses of the container?